### PR TITLE
Flockmind AI cleanup

### PR DESCRIPTION
**Pull request by BenLubar**
- remove sleeps in flockmind AI code
- remove lag_check from mob AI (this should only be used in spawned procs, not looping processes)
- fix flockbits sometimes targeting the same tile
- make monkeys treat the box that rips all your arms and legs off as a threat
- fix flockdrone logic for whether an enemy is caged
- fix flockminds being able to control other flocks' drones
- add flock ident packet for grilles and lattices
- inactive drones with AI disabled no longer count as "sentient" in the ident packet
- added task for AI drones in ident packet

Tested with 300 flockbits and 110 flockdrones and the server didn't die. Projectile spam seems to lag the chat box, but that's a different concern.

### CODE REVIEW

File: code/obj/flock/furniture.dm [hunk 1]
```diff
@@ -284,3 +292,11 @@
 		animate_flock_passthrough(mover)
 		return 1
 	return ..()
+
+/obj/grille/flock/special_desc(dist, mob/user)
+	if(isflock(user))
+		return {"<span class='flocksay'><span class='bold'>###=-</span> Ident confirmed, data packet received.
+		<br><span class='bold'>ID:</span> Reinforced Conduit
```
Comment (moonlol) [inline code/obj/flock/furniture.dm:299]: Sorry for the late review, I'd say change this to "Reinforced Barricade" instead of conduit.

**Review by moonlol — APPROVED** [review]

File: code/mob/living/critter/ai/flock/flocktasks.dm [hunk 2]
```diff
@@ -577,7 +568,7 @@
 		holder.target = get_best_target(get_targets())
 	if(holder.target)
 		var/mob/living/M = holder.target
-		if(M && !istype(M.loc.type, /obj/icecube/flockdrone) && (M.getStatusDuration("stunned") || M.getStatusDuration("weakened") || M.getStatusDuration("paralysis") || M.stat))
+		if(M && !istype(M.loc, /obj/icecube/flockdrone) && (M.getStatusDuration("stunned") || M.getStatusDuration("weakened") || M.getStatusDuration("paralysis") || M.stat))
```
Comment (ZeWaka) [inline code/mob/living/critter/ai/flock/flocktasks.dm:571]: Proc for all this iirc

File: code/mob/living/critter/ai/flock/flocktasks.dm [hunk 3]
```diff
@@ -104,13 +104,24 @@
 
 /datum/aiTask/sequence/goalbased/build/precondition()
 	. = 0
-	var/mob/living/critter/flock/drone/F = holder.owner
-	if(F?.can_afford(20))
+	var/mob/living/critter/flock/F = holder.owner
+	if (!F)
+		return
```
Comment (ZeWaka) [inline code/mob/living/critter/ai/flock/flocktasks.dm:109]: Return what

File: code/mob/living/critter/ai/flock/flocktasks.dm (hunk 2 above)
Comment (BenLubar) [reply code/mob/living/critter/ai/flock/flocktasks.dm]: Can't find one, but I see a `#define` in the cyborg code for "am I stunned"

Comment (ZeWaka) [reply code/mob/living/critter/ai/flock/flocktasks.dm]: it should be for all mobs

Comment (ZeWaka) [reply code/mob/living/critter/ai/flock/flocktasks.dm]: hrm it's not, can you make this into one? put in helpers.dm, a #define works

Comment (ZeWaka) [reply code/mob/living/critter/ai/flock/flocktasks.dm]: just the latter portion, not the ice cube stuff

File: code/procs/helpers.dm [hunk 4]
```diff
@@ -2643,3 +2643,10 @@ proc/keep_truthy(some_list)
 	for(var/x in some_list)
 		if(x)
 			. += x
+
+proc/has_incapacitating_status(mob/M)
```
Comment (ZeWaka) [inline code/procs/helpers.dm:2647]: weird name but ok

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**